hack/generated/controllers/deployment_reconciler_negative_test.go

```diff
@@ -0,0 +1,177 @@
+				Name:     to.StringPtr("STANDARD_D1_v2"),
+				Capacity: to.IntPtr(1),
+			},
+			PlatformFaultDomainCount: to.IntPtr(3),
+			SinglePlacementGroup:     to.BoolPtr(false),
+			UpgradePolicy: &compute.UpgradePolicy{
+				Mode: &upgradePolicyMode,
+			},
+			VirtualMachineProfile: &compute.VirtualMachineScaleSetVMProfile{
+				StorageProfile: &compute.VirtualMachineScaleSetStorageProfile{
+					ImageReference: &compute.ImageReference{
+						Publisher: to.StringPtr("this publisher"),
+						Offer:     to.StringPtr("does not"),
+						Sku:       to.StringPtr("exist"),
+						Version:   to.StringPtr("latest"),
+					},
+				},
+				OsProfile: &compute.VirtualMachineScaleSetOSProfile{
+					ComputerNamePrefix: to.StringPtr("computer"),
+					AdminUsername:      &adminUsername,
+				},
+				NetworkProfile: &compute.VirtualMachineScaleSetNetworkProfile{
+					NetworkInterfaceConfigurations: []compute.VirtualMachineScaleSetNetworkConfiguration{
+						{
+							Name: "mynicconfig",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// There are two ways that a deployment can fail. It can be rejected when initially
+// submitted to the Azure API, or it can be accepted and then report a failure during
+// long running operation polling. This ensures that the second case is handled correctly.
+func Test_DeploymentAccepted_LongRunningOperationFails(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+	rg := tc.CreateNewTestResourceGroupAndWait()
+	acct := newStorageAccountWithInvalidKeyExpiration(tc, rg)
+	tc.CreateResourceAndWaitForFailure(acct)
+
+	errAnnotation := acct.Annotations[reconcilers.ResourceErrorAnnotation]
+	tc.Expect(errAnnotation).To(ContainSubstring("InvalidValuesForRequestParameters"))
+	tc.Expect(errAnnotation).To(ContainSubstring("Values for request parameters are invalid: keyPolicy.keyExpirationPeriodInDays."))
+}
+
+func Test_DeploymentAccepted_LongRunningOperationFails_SucceedsAfterUpdate(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+	rg := tc.CreateNewTestResourceGroupAndWait()
+	acct := newStorageAccountWithInvalidKeyExpiration(tc, rg)
+	tc.CreateResourceAndWaitForFailure(acct)
+
+	// Remove the bad property and ensure we can successfully provision
+	patcher := tc.NewResourcePatcher(acct)
+	acct.Spec.KeyPolicy = nil
+	tc.PatchResourceAndWaitAfter(acct, patcher, armclient.FailedProvisioningState)
+
+	// Ensure that the old failure information was cleared away
+	objectKey, err := client.ObjectKeyFromObject(acct)
+	tc.Expect(err).ToNot(HaveOccurred())
+	updated := &storage.StorageAccount{}
+	tc.GetResource(objectKey, updated)
+
+	tc.Expect(updated.Annotations).ToNot(HaveKey(reconcilers.ResourceErrorAnnotation))
+}
+
+// There are two ways that a deployment can fail. It can be rejected when initially
+// submitted to the Azure API, or it can be accepted and then report a failure during
+// long running operation polling. This ensures that the first case is handled correctly.
```

Comment:
These comments on the intent of each test are really valuable, especially when there's a lot going on with the test; maybe add these to other tests in this file too?

Reply:
Done